pkg/cri/server/sandbox_run_linux.go

```diff
@@ -215,6 +215,10 @@ func (c *criService) sandboxContainerSpecOpts(config *runtime.PodSandboxConfig,
 		specOpts = append(specOpts, seccompSpecOpts)
 	}
 
+	if c.config.EnableCDI {
+		specOpts = append(specOpts, oci.WithCDI(config.Annotations, c.config.CDISpecDirs))
+	}
+
```

Comment:
I don't think you should inject CDI devices into a Pod's sandbox.

Reply:
why? And actually, If it is not injected into Pod's sandbox, it will cause the kata container to fail to start because no CDI device is passed in.

Reply:
> why? And actually, If it is not injected into Pod's sandbox, it will cause the kata container to fail to start because no CDI device is passed in.

Okay, my statement was a bit too hasty. Let me rephrase it so that AFAICT injecting devices into a Pod's sandbox was not explicitly considered (so far) among the use cases for CDI.

That said, I tried CDI-injecting a device into a kata-container using a containerd patched with the commit from this PR. It still failed (with an error like `QMP command failed: Could not open '/dev/vendor1-dev1': No such file or directory: unknown`).

Are there any other patches (for Kata-containers or containerd itself) that need to be in place for that injection to work ? Or was the problem that in my test I simply defined an arbitrary /dev entry (not corresponding to anything like a real GPU) and try to inject that into the Kata container ?

Reply:
@klihub Okay, perhaps I did not describe it clearly. Kata container support CDI is also in progress, ref PR: https://github.com/kata-containers/kata-containers/pull/4699. And I am testing it in combination with k8s and contained. if you want CDI-injecting a device into a kata-container using a containerd patched with the commit from this PR, pls patch this PR https://github.com/kata-containers/kata-containers/pull/4699 to kata.

Reply:
in k8s+containerd+kata-containers, I suppose the CDI annotation should be added to `sandbox anno`

Reply:
as from as I know, Pod annotation will be passed to the container runtime(contained) in `create Sandbox` interface instead of `Create Container` interface.